client/src/worldgen.rs

```diff
@@ -257,19 +264,59 @@ impl ChunkParams {
         }
     }
 
-    fn generate_road_support(&self, center: na::Vector3<f64>) -> Option<Material> {
+    fn generate_road_support(
+        &self,
+        center: na::Vector3<f64>,
+        coords: na::Vector3<u8>,
+    ) -> Option<Material> {
         let plane = -Plane::from(Side::B);
         let horizontal_distance = plane.elevation(center, self.chunk);
 
-        if horizontal_distance < 0.3 {
+        if (horizontal_distance < 0.3) && self.trussing_at(coords) {
             Some(Material::WoodPlanks)
         } else {
             None
         }
     }
 
+    /// Make a truss-shaped template
+    fn trussing_at(&self, coords: na::Vector3<u8>) -> bool {
+        //Generates planar diagonals, but corner is offset
+        let mut criteria_met = 0_u32;
+        let x = coords[0];
+        let y = coords[1];
+        let z = coords[2];
+
+        // straight lines.
+        if x == 2 * self.dimension / 3 {
+            criteria_met += 1;
+        }
+        if y == 2 * self.dimension / 3 {
+            criteria_met += 1;
+        }
+        if z == 2 * self.dimension / 3 {
```

Comment:
Let's compute this value once and store it in a temporary and reuse that.

Reply:
What scope should it have? trussing_at, generate_chunks, or global?

Reply:
I'm going to default to function-wide

Reply:
Good choice. In this case we're more concerned about readability/maintainability than efficiency, as the compiler can generally manage the latter fine at this scale.